### ISSUE TITLE
How do I test my trained model

### ISSUE THREAD

**Issue by kako523** [lllyasviel/ControlNet#304]
Hello! Your work is excellent! I trained my own data set with your method, but I don't know how to test the effect of my trained checkpoint, can I only check the results saved during training in the folder /image_log? Looking forward to your reply.

**Comment by mahdip72** [comment]
Same here. What is the difference between these images in the image_log:
reconstruction_gs-011070_e-000135_b-000000
samples_cfg_scale_9.00_gs-011070_e-000135_b-000000

**Comment by engrmusawarali** [comment]
Samples are predicted images and the reconstruction is the ground truth

**Comment by alelordelo** [comment]
hi @mahdip72  @kako523 @engrmusawarali ,

Did you guys fired out what exactly is" reconstruction" and "samples"?


**Image logs**
@engrmusawarali, are you 100% sure that  "reconstruction" is ground truth (dataset input data).?
Although both are very similar,  I do see a difference between my input and "reconstruction"....
My problem is that reconstruction images look fantastic, and samples look way too terrible and dont match anywhere close the input images...

**Saved model**
When training if finished, I dont find the model saved to the models folder....

**Comment by alelordelo** [comment]
thanks for the detailed explanation  @mahdip72 !

did you manage to get your model properly trained?

**Comment by HassanBinHaroon** [comment]
How to use your own trained model?

**Comment by YunChangLi** [comment]
same

**Comment by Khaihuyennguyen** [comment]
> @YunChangLi I have the idea but I just want to confirm.
> 
> My model is still on training. After the training completion, I'll try to execute that.

Do you have any update? I have trained my model, now I want to use the train model to test

**Comment by YunChangLi** [comment]
> > @YunChangLi I have the idea but I just want to confirm.
> > My model is still on training. After the training completion, I'll try to execute that.
> 
> Do you have any update? I have trained my model, now I want to use the train model to test

I have showed the correct output with my testing code. The issue that occurred earlier was related to color space thresholds. After correction, my results turned out perfectly. Thanks for your reply

**Comment by HassanBinHaroon** [comment]
Training and testing configuration should be same for training and testing.
Deep models do depend upon training circumstances. If you test with
different (or sometimes slightly) circumstances, then the results can be
different.

**Comment by KhawlahB** [comment]
> > > @YunChangLi I have the idea but I just want to confirm.
> > > My model is still on training. After the training completion, I'll try to execute that.
> > 
> > 
> > Do you have any update? I have trained my model, now I want to use the train model to test
> 
> I have showed the correct output with my testing code. The issue that occurred earlier was related to color space thresholds. After correction, my results turned out perfectly. Thanks for your reply

How you test your trained model? i could not see any script for inference... i want to evaluate my trained model on the validation set.

looking for your reply please... @YunChangLi @mahdip72 @Khaihuyennguyen @HassanBinHaroon

**Comment by Palavra1** [comment]
hello,did you solve the problem? I have the same problem and didn't know how to solve it :(
